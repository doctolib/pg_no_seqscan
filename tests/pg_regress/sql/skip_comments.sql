-- Test ignoring seqscans with skip comments

SET pg_no_seqscan.level = ERROR;
CREATE TABLE test_skip AS (SELECT * FROM generate_series(1,10) AS id);

EXPLAIN SELECT * FROM test_skip;
SELECT * FROM test_skip; -- should fail

-- Test with skip comment variations
SELECT * FROM test_skip /* pg_no_seqscan_skip */;
SELECT * FROM test_skip /* host_name:a-b-1.2.foo,db:my_database,git:0123456789abcdef,pg_no_seqscan_skip,path:/foo/source.java:108`(<>)' */;
SELECT * FROM test_skip /*pg_no_seqscan_skip*/;

DROP TABLE test_skip;
