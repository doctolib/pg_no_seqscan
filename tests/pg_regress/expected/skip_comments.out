-- Test ignoring seqscans with skip comments
SET pg_no_seqscan.level = ERROR;
CREATE TABLE test_skip AS (SELECT * FROM generate_series(1,10) AS id);
EXPLAIN SELECT * FROM test_skip;
                         QUERY PLAN                          
-------------------------------------------------------------
 Seq Scan on test_skip  (cost=0.00..35.50 rows=2550 width=4)
(1 row)

SELECT * FROM test_skip; -- should fail
ERROR:  A 'Sequential Scan' on test_skip has been detected.
  - Run an EXPLAIN on your query to check the query plan.
  - Make sure the query is compatible with the existing indexes.

Query: SELECT * FROM test_skip;

-- Test with skip comment variations
SELECT * FROM test_skip /* pg_no_seqscan_skip */;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

SELECT * FROM test_skip /* host_name:a-b-1.2.foo,db:my_database,git:0123456789abcdef,pg_no_seqscan_skip,path:/foo/source.java:108`(<>)' */;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

SELECT * FROM test_skip /*pg_no_seqscan_skip*/;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

DROP TABLE test_skip;
